<a href="https://colab.research.google.com/github/alex-s-coffee/main/blob/main/Forbes_Richest_Athletes_1990_2020_EDA11.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
# Importing the required packages 
import pandas as pd
import plotly.express as px


In [2]:
# Load the dataset and create a dataframe for the analysis

ath_df= pd.read_csv('Forbes Richest Atheletes (Forbes Richest Athletes 1990-2020).csv')
ath_df[:]

S.NO               Name Nationality  Current Rank Previous Year Rank  \
0       1         Mike Tyson         USA             1                NaN   
1       2     Buster Douglas         USA             2                NaN   
2       3  Sugar Ray Leonard         USA             3                NaN   
3       4       Ayrton Senna      Brazil             4                NaN   
4       5        Alain Prost      France             5                NaN   
..    ...                ...         ...           ...                ...   
296   297      Stephen Curry         USA             6                  9   
297   298       Kevin Durant         USA             7                 10   
298   299        Tiger Woods         USA             8                 11   
299   300       Kirk Cousins         USA             9               >100   
300   301       Carson Wentz         USA            10               >100   

                 Sport  Year  earnings ($ million)  
0               boxing  1990                  28.6  
1               boxing  1990                  26.0  
2               boxing  1990                  13.0  
3          auto racing  1990                  10.0  
4          auto racing  1990                   9.0  
..                 ...   ...                   ...  
296         Basketball  2020                  74.4  
297         Basketball  2020                  63.9  
298               Golf  2020                  62.3  
299  American Football  2020                  60.5  
300  American Football  2020                  59.1  

[301 rows x 8 columns]

## Cleaning

In [3]:
# Finding all null values and data type for each column

ath_df.isnull().sum(), ath_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 301 entries, 0 to 300
Data columns (total 8 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   S.NO                  301 non-null    int64  
 1   Name                  301 non-null    object 
 2   Nationality           301 non-null    object 
 3   Current Rank          301 non-null    int64  
 4   Previous Year Rank    277 non-null    object 
 5   Sport                 301 non-null    object 
 6   Year                  301 non-null    int64  
 7   earnings ($ million)  301 non-null    float64
dtypes: float64(1), int64(3), object(4)
memory usage: 18.9+ KB


(S.NO                     0
 Name                     0
 Nationality              0
 Current Rank             0
 Previous Year Rank      24
 Sport                    0
 Year                     0
 earnings ($ million)     0
 dtype: int64,
 None)

All columns are in the right format, apart from the 'Previous Year Rank' column , which is also the only one that has null values

Witha further investigation we can see that some values in this column are 'none' and '??' and some are too vague like '>20'. I will convert the column to numeric and the strings that can't be converted will be turned into null values.  

In [4]:
# Convert to numeric and count the null values

pd.to_numeric(ath_df['Previous Year Rank'],errors='coerce').isnull().sum()

89

The 89 missing values in 'Previous Year Rank' account for almost 30% of the total.

It's also very difficult to fill in the missing values, as there's close to nothing in the dataset that we could infer it from.  
Unless I research it manually, but that would be counter productive.

So I am definitely confident to drop it. 

In [5]:
ath_df.drop(['Previous Year Rank'], inplace=True, axis=1)
ath_df[:]

S.NO               Name Nationality  Current Rank              Sport  \
0       1         Mike Tyson         USA             1             boxing   
1       2     Buster Douglas         USA             2             boxing   
2       3  Sugar Ray Leonard         USA             3             boxing   
3       4       Ayrton Senna      Brazil             4        auto racing   
4       5        Alain Prost      France             5        auto racing   
..    ...                ...         ...           ...                ...   
296   297      Stephen Curry         USA             6         Basketball   
297   298       Kevin Durant         USA             7         Basketball   
298   299        Tiger Woods         USA             8               Golf   
299   300       Kirk Cousins         USA             9  American Football   
300   301       Carson Wentz         USA            10  American Football   

     Year  earnings ($ million)  
0    1990                  28.6  
1    1990                  26.0  
2    1990                  13.0  
3    1990                  10.0  
4    1990                   9.0  
..    ...                   ...  
296  2020                  74.4  
297  2020                  63.9  
298  2020                  62.3  
299  2020                  60.5  
300  2020                  59.1  

[301 rows x 7 columns]

In [6]:
# Let's now drop all the duplicates
ath_df.drop_duplicates(inplace=True)
ath_df.shape[0]

301

NO duplicates, which means one year has 11 entries

Let's find out


In [7]:
year = ath_df.groupby('Year')
year_count = year['Year'].count()
year_count.sort_values()[-5:]


Year
1992    10
1991    10
2019    10
2020    10
2002    11
Name: Year, dtype: int64

And last we'll check that all names of athletes and sports are unique and there aren't misspellings

In [8]:
# let's start with the athletes' names
names = ath_df['Name'].unique()
names.sort()
names

array(['Aaron Rodgers', 'Aaron Rogers', 'Alain Prost', 'Alex Rodriguez',
       'Andre Agassi', 'Andrew Luck', 'Arnold Palmer', 'Ayrton Senna',
       'Buster Douglas', 'Cam Newton', 'Canelo Alvarez', 'Carson Wentz',
       'Cecil Fielder', 'Conor McGregor', 'Cristiano Ronaldo',
       'Dale Earnhardt', 'Dale Earnhardt Jr.', 'David Beckham',
       'Deion Sanders', 'Dennis Rodman', 'Donovan "Razor" Ruddock',
       'Drew Brees', 'Eli Manning', 'Emmit Smith', 'Evander Holyfield',
       'Floyd Mayweather', 'Gary Sheffield', 'George Foreman',
       'Gerhard Berger', 'Grant Hill', 'Greg Norman', 'Jack Nicklaus',
       'Jacques Villeneuve', 'James Harden', 'Jeff Gordon', 'Jim Courier',
       'Joe Montana', 'Joe Sakic', 'Jordan Spieth', 'Kevin Durant',
       'Kevin Garnett', 'Kimi Raikkonen', 'Kirk Cousins', 'Kobe Bryant',
       'Lance Armstrong', 'LeBron James', 'Lennox Lewis',
       'Lewis Hamilton', 'Lionel Messi', 'Manny Pacquiao', 'Matt Ryan',
       'Matthew Stafford', 'Michael 

Aaron Rodgers and Shaquille O'Neal have two different spellings

In [9]:
# Replace the misspellings
ath_df['Name'] = ath_df['Name'].replace('Aaron Rogers','Aaron Rodgers')
ath_df['Name'] = ath_df['Name'].replace("Shaquille O'Neal","Shaq O'Neal")

names = ath_df['Name'].unique()
names.sort()
names

array(['Aaron Rodgers', 'Alain Prost', 'Alex Rodriguez', 'Andre Agassi',
       'Andrew Luck', 'Arnold Palmer', 'Ayrton Senna', 'Buster Douglas',
       'Cam Newton', 'Canelo Alvarez', 'Carson Wentz', 'Cecil Fielder',
       'Conor McGregor', 'Cristiano Ronaldo', 'Dale Earnhardt',
       'Dale Earnhardt Jr.', 'David Beckham', 'Deion Sanders',
       'Dennis Rodman', 'Donovan "Razor" Ruddock', 'Drew Brees',
       'Eli Manning', 'Emmit Smith', 'Evander Holyfield',
       'Floyd Mayweather', 'Gary Sheffield', 'George Foreman',
       'Gerhard Berger', 'Grant Hill', 'Greg Norman', 'Jack Nicklaus',
       'Jacques Villeneuve', 'James Harden', 'Jeff Gordon', 'Jim Courier',
       'Joe Montana', 'Joe Sakic', 'Jordan Spieth', 'Kevin Durant',
       'Kevin Garnett', 'Kimi Raikkonen', 'Kirk Cousins', 'Kobe Bryant',
       'Lance Armstrong', 'LeBron James', 'Lennox Lewis',
       'Lewis Hamilton', 'Lionel Messi', 'Manny Pacquiao', 'Matt Ryan',
       'Matthew Stafford', 'Michael Jordan', 'Michae

In [10]:
# Check the sports' names
sports = ath_df['Sport'].unique()
sports.sort()
sports

array(['American Football', 'American Football / Baseball', 'Auto Racing',
       'Auto Racing (Nascar)', 'Auto racing', 'Baseball', 'Basketball',
       'Boxing', 'F1 Motorsports', 'F1 racing', 'Golf', 'Hockey',
       'Ice Hockey', 'MMA', 'NASCAR', 'NBA', 'NFL', 'Soccer', 'Tennis',
       'auto racing', 'baseball', 'basketball', 'boxing', 'cycling',
       'golf', 'ice hockey', 'motorcycle gp', 'soccer', 'tennis'],
      dtype=object)

In [11]:
ath_df['Sport'] = ath_df['Sport'].str.lower()
sports = ath_df['Sport'].unique()
sports.sort()
sports

array(['american football', 'american football / baseball', 'auto racing',
       'auto racing (nascar)', 'baseball', 'basketball', 'boxing',
       'cycling', 'f1 motorsports', 'f1 racing', 'golf', 'hockey',
       'ice hockey', 'mma', 'motorcycle gp', 'nascar', 'nba', 'nfl',
       'soccer', 'tennis'], dtype=object)

In [12]:
# let's do the necessary replacements
ath_df['Sport'] = ath_df['Sport'].replace('american football / baseball','american football')
ath_df['Sport'] = ath_df['Sport'].replace('auto racing','f1 racing')
ath_df['Sport'] = ath_df['Sport'].replace('f1 motorsports','f1 racing')
ath_df['Sport'] = ath_df['Sport'].replace('auto racing (nascar)','nascar')
ath_df['Sport'] = ath_df['Sport'].replace('nfl','american football')
ath_df['Sport'] = ath_df['Sport'].replace('nba','basketball')

sports = ath_df['Sport'].unique()
sports.sort()
sports

array(['american football', 'baseball', 'basketball', 'boxing', 'cycling',
       'f1 racing', 'golf', 'hockey', 'ice hockey', 'mma',
       'motorcycle gp', 'nascar', 'soccer', 'tennis'], dtype=object)

# EDA

In [13]:
ath_df.describe()

S.NO  Current Rank         Year  earnings ($ million)
count  301.000000    301.000000   301.000000            301.000000
mean   151.000000      5.448505  2005.122924             45.516279
std     87.035433      2.850995     9.063563             33.525337
min      1.000000      1.000000  1990.000000              8.100000
25%     76.000000      3.000000  1997.000000             24.000000
50%    151.000000      5.000000  2005.000000             39.000000
75%    226.000000      8.000000  2013.000000             59.400000
max    301.000000     10.000000  2020.000000            300.000000

The average earning per athlete per year is 45.5 mil, which means every year on average has 455 mil earnings. 

Let's track the earnings year by year and see if there is a trend. 

In [14]:
earnings_by_year = ath_df.groupby(['Year'])['earnings ($ million)'].sum().reset_index().sort_values('Year', ascending=False)


fig = px.line(earnings_by_year, x = 'Year', y = 'earnings ($ million)', title= 'Total earnings each year', labels={"Year": "Year", "earnings ($ million)": "Total earnings (in US$ million)"})
fig.show()


We can definitely see a growing trend, with two outstanding years in 2015 and 2018. 

Let's take a closer look

In [15]:
top_earning =ath_df.sort_values(['earnings ($ million)'],ascending=False).groupby('Year').head(1)

fig = px.bar(top_earning, x ='Year', y = 'earnings ($ million)', color='Name', title= '#1 earner each year', labels={"Year": "Year", "earnings ($ million)": " #1 earnings (in US$ million)","Name":"Athlete"})
fig.show()



The average top earning score up to the year 2000 is aroud 50 million. 

After that it doubles to an average of 100 mil for the top position, but 2015 and 2018 have both triple that amount, with two outstanding scores from Floyd Maywether. That surely is one of the reasons that explains the amazing perfomances of the two years. 

#### Which athlete has most top10 appearances?

In [16]:
appearance_count = ath_df.groupby('Name').count().reset_index().sort_values(by='S.NO', ascending= False)[:20]

fig = px.bar(appearance_count, x ='S.NO', y = 'Name', color='Name', title= 'Most top10 appearances (top 20)', labels={"S.NO": "top10 appearances","Name":"Athlete"},height=600)
fig.show()



Tiger Woods and Michael Jordan have the most appearances with 19, but Tiger Woods has toped the charts 11 times out of those 19, while Michael Jordan only 6 (see previous graph), Kobe Bryant and Shaq O'Neal follow in 3rd and 4th making it a very basketball-heavy top5. 

#### Which athlete has made the most? 

In [17]:
earnings_per_player = ath_df.groupby('Name').sum().reset_index().sort_values(by= 'earnings ($ million)', ascending=False)[:20]
earnings_per_player

fig = px.bar(earnings_per_player, x ='earnings ($ million)', y = 'Name', color='Name', title= 'Top career earnings (top 20)', labels={"Name":"Athlete"},height=600)
fig.show()


<ipython-input-17-1221b4250ab6>:1: FutureWarning:

The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.



Tiger Woods tops the charts when it comes to career earnings as well. 

LeBron James is the richest basketball player, although Michael Jordan and Kobe Bryant have both more appearancesa than him. 

Floyd Mayweather is the biggest surprise of this chart: he is 3rd with only 5 top10 appearances.

#### Which is the most lucrative sport?

In [18]:
earnings_per_sport = ath_df.groupby('Sport').sum().reset_index().sort_values(by= 'earnings ($ million)', ascending=False)

fig = px.pie(earnings_per_sport, values='earnings ($ million)', names='Sport', title='Most lucrative sports')
fig.show()

<ipython-input-18-f53b1b779a46>:1: FutureWarning:

The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.



One sport is the biggest industry by far, and that is basketball. 

And 4 sports make for almost 75% of the top10 earnings over 30 years. That's quite impressive!

My biggest surprise here was to see that boxing and golf are just as lucrative as soccer. 
While soccer is wildly popular but a cheap sport, golf and boxing are niches where followers are ready to pay a premium and create very good profits for the athletes performing in them.   